src/components/SelectionList/BaseSelectionList.tsx

```diff
@@ -286,6 +295,11 @@ function BaseSelectionList<TItem extends ListItem>(
         selectRow(focusedOption);
     };
 
+    // This debounce happens on the trailing edge because on repeated enter presses, rapid component state update cancels the existing debounce and the redundant
+    // enter presses runs the debounced function again. This is solved by running the debounce on trailing edge.
```

Comment:
I feel like the last sentence is redundant given we mention it already at the start of the comment

```suggestion
    // This debounce happens on the trailing edge because on repeated enter presses, rapid component state update cancels the existing debounce and the redundant
    // enter presses runs the debounced function again.
```